src/emu/services/include/services/window/common.h

```diff
@@ -685,4 +686,6 @@ namespace eka2l1::epoc {
     TKeyCode map_scancode_to_keycode(TStdScanCode scan_code);
 
     TStdScanCode map_inputcode_to_scancode(int input_code, int ui_rotation);
+    std::uint32_t map_button_to_inputcode(std::map<std::pair<int, int>, std::uint32_t> &map, int controller_id, int button);
```

Comment:
I did not expect you to pull off putting a map as a reference :D. No problem, but give it an alternative typename is recommended.